hpedockerplugin/file_manager.py

```diff
@@ -540,19 +516,29 @@ def _create_share(self, share_name, share_args):
 
         # Make copy of args as we are going to modify it
         fpg_name = share_args.get('fpg')
+        undo_cmds = []
 
-        if fpg_name:
-            self._create_share_on_fpg(
-                share_args,
-                self._get_existing_fpg,
-                self._create_fpg
-            )
-        else:
-            self._create_share_on_fpg(
-                share_args,
-                self._get_default_available_fpg,
-                self._create_default_fpg
-            )
+        try:
+            if fpg_name:
+                self._create_share_on_fpg(
+                    share_args,
+                    self._get_existing_fpg,
+                    self._create_fpg,
+                    undo_cmds
+                )
+            else:
+                self._create_share_on_fpg(
+                    share_args,
+                    self._get_default_available_fpg,
+                    self._create_default_fpg,
+                    undo_cmds
+                )
+        except exception.PluginException as ex:
```

Comment:
@imran-ansari Here we are not expecting ShareCreationFailed exception. _create_share_on_fpg is finally raising ShareCreationFailed exception.
Hence this also has to be caught and status should be changed to failed.

Reply:
PluginException is the base class of all our custom exceptions. Hence it will catch ShareCreationFailed as well as any child of PluginException here.